src/matcher/index.ts

```diff
@@ -71,11 +74,66 @@ export function createRouterMatcher(
     return matcherMap.get(name)
   }
 
+  function routeComponentsOptionGuard(record: RouteRecordRaw): never | void {
```

Comment:
the function should just receive the `components` option and any other relevant option like name or path and not be ran if other scenarios

Reply:
Thanks for review!

Shouldn't we use `path`? 

```javascript
    warn(
            `Components property in record  with path 👉 "${path}" is not valid. ` +
              `Property "components" should be object. ` +
              `ex> { components: { default: Home } }`
          )
```

I think showing path we be more helpful debug codes.